engine/storage/snapshot/src/org/apache/cloudstack/storage/snapshot/SnapshotObject.java

```diff
@@ -149,6 +149,17 @@ public long getPhysicalSize() {
         return physicalSize;
     }
 
+    @Override
+    public boolean markBackedUp() {
```

Comment:
Why returning a boolean here?

I mean, you capture a checked exception here, and then you return `false`. Then, in the method where you use this `markBackedUp`, you will check if the return is false, and then you throw a runtime exception. Why not capture the checked exception and throw the runtime here?

Also, what about unit test cases?

Reply:
Mainly I just couldn't throw a NoTransitionException because it would create a circular dependency.  CloudRuntimeException would work though, since as you say that's ultimately what ends up happening in the one place this gets called (for now).  I have no problem making this change.

Reply:
I am +1 on the @rafaelweingartner idea. Throwing a CloudRuntimeException would be better than the boolean approach.